docs/design/undo-redo.md

```
@@ -569,3 +579,99 @@ d1 redo: insert "X" via CRDTTreePos → same position on all clients
 → Convergence restored
 ```
+
+### Analysis: Array Set + Move Undo
+
+When an element is moved and then set (value replaced), undoing the set
+restores the value at the element's original dead position instead of
+its moved position.
+
+#### Concrete Example
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)